### PR TITLE
<style> Tailwind CSS syntax prevents HMR from not working.

### PR DESCRIPTION
**Pull request by L4Ph**
The following issue has been resolved.
This is a temporary solution, and we are considering further revisions and changes later.

https://github.com/saicaca/fuwari/issues/192

### CODE REVIEW

**Comment by L4Ph** [comment]
The banner image seems to have disappeared, so I'll investigate a bit... 😢

**Comment by thedevstone** [comment]
> The banner image seems to have disappeared, so I'll investigate a bit... 😢

In the PR you commented out the banner?

**Comment by L4Ph** [comment]
```js
  banner: {
    enable: false,
  }
```

...
but image cannot load.
We will continue investigating.

**Comment by L4Ph** [comment]
image
🤔

**Comment by thedevstone** [comment]
> image 🤔

cloned the PR, confirmed that is not working

**Comment by thedevstone** [comment]
Fixed:
Screenshot from 2024-09-27 11-52-29
Cannot push, please @L4Ph fix this according to photo, thanks

File: src/layouts/Layout.astro [hunk 1]
```diff
@@ -151,44 +149,14 @@ const bannerOffset = bannerOffsetByPosition[siteConfig.banner.position || 'cente
...
 	</body>
 </html>
+
 <style is:global>
 	:root {
 		--hue: var(--configHue);
 		--page-width: 75rem;
 	}
 </style>
-<style is:global define:vars={{ bannerOffset }}>
```
Comment (saicaca) [inline src/layouts/Layout.astro:160]: This line defines the CSS variable `--bannerOffset`. Removing this causes it to become undefined. I am not sure but this might be the reason why the banner isn't working.

File: src/layouts/Layout.astro [hunk 2]
```diff
@@ -130,8 +130,6 @@ const bannerOffset = bannerOffsetByPosition[siteConfig.banner.position || 'cente
 
 		<link rel="alternate" type="application/rss+xml" title={profileConfig.name} href={`${Astro.site}rss.xml`}/>
 
-		<style define:vars={{ configHue }}></style>  <!-- defines global css variables. This will be applied to <html> <body> and some other elements idk why -->
```
Comment (saicaca) [inline src/layouts/Layout.astro:133]: Here too. I suggest keeping this style element to store the CSS variables (moving `bannerOffset` here as well).

Comment (thedevstone) [reply src/layouts/Layout.astro:133]: removing `opacity: 0` from ImageWrapper fixed the issue. But i'm not very into the project so i'm probably wrong

File: src/layouts/Layout.astro (hunk 1 above)
Comment (L4Ph) [reply src/layouts/Layout.astro:160]: resolve

File: src/layouts/Layout.astro (hunk 2 above)
Comment (L4Ph) [reply src/layouts/Layout.astro]: resolve

**Comment by thedevstone** [comment]
Please merge the fix cos it's a very frustrating having the HMR not working

File: .frontmatter/database/taxonomyDb.json
Comment (saicaca) [inline .frontmatter/database/taxonomyDb.json:1]: Please remove this file.

File: src/base.css [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+        @apply opacity-0
+    }
+    .copy-btn.success .success-icon {
+        @apply opacity-100
+    }
+}
+
+@layer components {
+    @layer components {
+        .dash-line {
```
Comment (saicaca) [inline src/base.css:147]: There are two `@layer` wrapping here. Is this a mistake?

File: src/layouts/Layout.astro [hunk 4]
```diff
@@ -151,17 +151,15 @@ const bannerOffset = bannerOffsetByPosition[siteConfig.banner.position || 'cente
...
 		--page-width: 75rem;
 	}
 </style>
-<style is:global define:vars={{ bannerOffset }}>
-@tailwind components;
-@tailwind utilities;
 
-@layer components {
+<style is:global define:vars={{ bannerOffset }}>
 	.enable-banner.is-home #banner-wrapper {
```
Comment (saicaca) [inline src/layouts/Layout.astro:163]: The CSS code here needs to be wrapped inside `@layer components` to ensure `lg:is-home` is properly processed, which is needed for the banner height change on the homepage.

File: src/config.ts [hunk 5]
```diff
@@ -15,7 +15,7 @@ export const siteConfig: SiteConfig = {
     fixed: false,     // Hide the theme color picker for visitors
   },
   banner: {
-    enable: false,
+    enable: true,
```
Comment (saicaca) [inline src/config.ts:18]: Please revert the change here.

File: src/layouts/Layout.astro (hunk 4 above)
Comment (L4Ph) [reply src/layouts/Layout.astro:163]: https://github.com/tailwindlabs/tailwindcss/discussions/5934#discussioncomment-2793112

I have a similar problem, and `@layer components` is not working.

File: src/base.css (hunk 3 above)
Comment (L4Ph) [reply src/base.css:147]: resolve

File: src/config.ts (hunk 5 above)
Comment (L4Ph) [reply src/config.ts:18]: resolve

File: src/layouts/Layout.astro (hunk 4 above)
Comment (L4Ph) [reply src/layouts/Layout.astro:163]: I was just forgetting `@tailwind components;`.

**Comment by thedevstone** [comment]
News on this PR?

**Comment by saicaca** [comment]
image

@L4Ph I don't know why, but the `max-w-none` here seems to be overridden. Prepending a `!` should solve the problem. `!max-w-none`

https://github.com/saicaca/fuwari/blob/8beeba81db3a5492aef52b712b47275013c74935/src/components/misc/Markdown.astro#L10

**Comment by saicaca** [comment]
Let me fix it. Thank you for your contribution.

**Comment by L4Ph** [comment]
Thanks!
